packages/venia-concept/src/components/Header/header.js

```diff
@@ -44,21 +55,32 @@ class Header extends Component {
                         </NavTrigger>
                     </div>
                     <div className={classes.secondaryActions}>
-                        <button className={classes.searchTrigger}>
-                            <Icon name="search" />
-                        </button>
+                        <SearchTrigger
+                            searchOpen={searchOpen}
+                            toggleSearch={toggleSearch}
+                        >
+                            {this.searchIcon}
+                        </SearchTrigger>
+                        <Route
+                            exact
+                            path="/search.html"
+                            render={() => {
+                                const { searchOpen, toggleSearch } = this.props;
+                                if (!searchOpen) {
+                                    toggleSearch();
+                                }
```

Comment:
> Mutations, subscriptions, timers, logging, and other side effects are not allowed inside the main body of a function component (referred to as React’s render phase). Doing so will lead to confusing bugs and inconsistencies in the UI.

We don't dispatch actions from a `render` function. Here we're actually in *two* render functions. 😅 

This needs to be moved to a lifecycle method.